app/src/main/java/com/example/mytestproject/ui/weatherData/todayWeather/CurrentWeatherFragment.kt

```diff
@@ -23,26 +23,27 @@ class CurrentWeatherFragment : Fragment() {
     lateinit var currentWeatherFactory: CurrentWeatherFactory
     private lateinit var currentWeatherViewModel: CurrentWeatherViewModel
 
-    private lateinit var weatherDataBinding: WeatherDataFragmentBinding
+    private lateinit var weatherDataBinding: CurrentWeatherDataFragmentBinding
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
 
         weatherDataBinding = DataBindingUtil.inflate(
-            layoutInflater, R.layout.weather_data_fragment, container, false
+            layoutInflater, R.layout.current_weather_data_fragment, container, false
         )
 
         return weatherDataBinding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        setHasOptionsMenu(true)
 
         setTitle()
 
-        val animationDrawable = weather_data_fragment.background as AnimationDrawable
+        val animationDrawable = weather_data_fragment.background as AnimationDrawable //screen background animation
```

Comment:
Can be simplified with animationDrawable.run { ... }

Reply:
Fixed